### PR TITLE
STM Readme update

### LINKED ISSUE #615: Revise the Readme of stm
## Why

`README.md` of the `mithril-stm` contains outdated information and its content needs to be improved.

## What

The readme content needs to be revised to provide a clear and informative explanation of the stm crate.

### PR DESCRIPTION
**Pull request by curiecrypt**
## Content

This PR includes the updates in the `README.md` file of **mithril-stm**.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [x] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## ToDo
- [x] Update the outdated details
- [x] Expand the explanation
- [x] Spell-check

## Issue(s)

Closes #615

### CODE REVIEW

File: mithril-stm/README.md [hunk 1]
```diff
@@ -91,50 +124,56 @@ fn main() {
...
-You can run tests of the library using `cargo test` (we recommend to use the `--release` flag, otherwise
-the tests might take a while) and run benchmarks using `cargo bench`. This crate uses `criterion` to run
-benchmarks.
+## Benchmarks
 
-We have run the benchmarks on an Apple M1 Pro machine with 16 GB of RAM, on macOS 12.6.
+Here we give the benchmark results of STM for size and time. We run the benchmarks on macOS 12.6 on an Apple M1 Pro machine with 16 GB of RAM.
 
-Note that single signatures in batch compat version does not depend on any variable and size of an individual signature is `176` bytes.
+Note that single signatures in batch compatible version do not depend on any variable and **the size of an individual signature is 176 bytes**.
```
Comment (iquerejeta) [inline mithril-stm/README.md:147]: The constant size of individual signatures is not tied to the batch proofs. Instead, it is tied to the fact that we simply do not include merkle paths in individual signatures, I would suggest to change this sentence as: 

`Note that single signatures do not contain the Merkle path, and therefore have a **constant size of 176 bytes**.`

**Review by iquerejeta — APPROVED** [review]
Looks good. Some minor comments, that once addressed, it can be merged.

File: mithril-stm/README.md [hunk 2]
```diff
@@ -1,40 +1,73 @@
...
+    * Key registration procedure for STM signatures
+    * The tests for the library functions and STM scheme
+    * Benchmark tests
+
+## Pre-requisites
+
+**Install Rust**
+
+* Install a [correctly configured](https://www.rust-lang.org/learn/get-started) Rust toolchain (latest stable version).
+* Install Rust [Clippy](https://github.com/rust-lang/rust-clippy) component.
```
[the remark this answers is not in the document]
Comment (jpraynaud) [reply mithril-stm/README.md:24]: It is not mandatory to build and run the library :+1:

File: mithril-stm/README.md (hunk 1 above)
Comment (iquerejeta) [reply mithril-stm/README.md]: Actually, signatures don't have constant size, right? They depend on how many indices they've won. @curiecrypt

**Pull request merged**